core/new-gui/src/app/workspace/component/workflow-editor/comment-box-modal/nz-modal-comment-box.component.ts

```diff
@@ -0,0 +1,56 @@
+import { Component, Input, OnInit } from "@angular/core";
+import { FormBuilder, Validators } from "@angular/forms";
+import { NzModalRef } from "ng-zorro-antd/modal";
+import { CommentBox, Comment } from "src/app/workspace/types/workflow-common.interface";
+import { WorkflowActionService } from "src/app/workspace/service/workflow-graph/model/workflow-action.service";
+import { UserService } from "src/app/common/service/user/user.service";
+import { User } from "src/app/common/type/user";
+import { untilDestroyed } from "@ngneat/until-destroy";
+import { UntilDestroy } from "@ngneat/until-destroy";
+
+@UntilDestroy()
+@Component({
+  selector: "texera-ngbd-modal-comment-box",
+  templateUrl: "./nz-modal-comment-box.component.html",
+  styleUrls: ["./nz-modal-comment-box.component.scss"],
+})
+export class NzModalCommentBoxComponent {
+  @Input() commentBox!: CommentBox;
+
+  public user: User | undefined;
+
+  public commentForm = this.formBuilder.group({
+    comment: ["", [Validators.required]],
+  });
+  constructor(
+    public workflowActionService: WorkflowActionService,
+    private formBuilder: FormBuilder,
+    public userService: UserService,
+    public modal: NzModalRef<any, number>
+  ) {
+    this.userService
+      .userChanged()
+      .pipe(untilDestroyed(this))
+      .subscribe(user => (this.user = user));
+  }
+
+  public onClickAddComment(): void {
+    if (this.commentForm.get("comment")?.invalid) {
+      alert("Cannot Submit Empty Comment!!");
+      return;
+    }
+    const newComment = this.commentForm.get("comment")?.value;
+    this.updateComments(newComment);
+    this.commentForm.get("comment")?.setValue("");
+  }
+
+  public updateComments(newComment: string): void {
+    const currentTime: string = new Date().toISOString();
+    const creator = this.user?.name;
```

Comment:
here I am hesitate, whether we should use `user name` or `user id`. Certainly user name should be used to display on the frontend, but the data we save into backend db should contain user id as that's the unique identifier. Note that there exists users with the same name.

Reply:
Now has two properties: creator for frontend display, creatorID for user identification.